Домашнее задание спринт 16

NekhoroshkinaElena opened the pull request.

NekhoroshkinaElena committed:
Разнести классы по пакетам

NekhoroshkinaElena committed:
Переписать код с использованием шаблона MVVM

NekhoroshkinaElena committed:
Переименовать классы

taikonavt commented on app/src/main/java/com/example/playlistmaker/settings/ui/view_model/SettingsViewModel.kt:
```diff
@@ -0,0 +1,61 @@
+        }
+    }
+
+    fun getStateThemeApp(): MutableLiveData<ThemeState> = stateThemeApp
```
отдаешь в активити MutableLiveData. Закрой интерфейсом LiveData

taikonavt commented on app/src/main/java/com/example/playlistmaker/player/ui/activity/TrackActivity.kt:
```diff
@@ -1,51 +1,44 @@
 import com.example.playlistmaker.util.getYearFromString
 import com.example.playlistmaker.util.millisecondToMinute
 
 const val IMAGE_CORNER_RADIUS = 8
```
на макетах размеры указываются в dp. Glide принимает размер скругления в фактических пикселях устройства. Поэтому нам нужно выполнить преобразование из dp в px. Как это сделать можно посмотреть в лекции по glide

taikonavt replied:
если константа не используется в другом классе, то лучше сразу закрывать видимость private

taikonavt commented on app/src/main/java/com/example/playlistmaker/creator/App.kt:
```diff
@@ -35,10 +35,7 @@ class App : Application() {
-            return true
-        }
-        return false
+        return sharedPrefs.contains(THEME_SWITCH_KEY)
```
эту логику sharedPrefs тоже нужно дергать из data слоя

NekhoroshkinaElena committed:
fix: исправить MutableLiveData на LiveData
fix: преобразовать dp в px, для Glide
fix: перенести логику работы с sharedPreferences в data слой

taikonavt commented on app/src/main/java/com/example/playlistmaker/settings/ui/view_model/SettingsViewModel.kt:
```diff
@@ -50,11 +56,12 @@ class SettingsViewModel(
-    }
-
     fun updateTheme(isChecked: Boolean) {
+        if (isChecked) {
+            AppCompatDelegate.setDefaultNightMode(AppCompatDelegate.MODE_NIGHT_YES)
+        } else {
+            AppCompatDelegate.setDefaultNightMode(AppCompatDelegate.MODE_NIGHT_NO)
+        }
```
как вариант, чтобы не дублировать код по установке темы здесь и в App, можно оставить код в Арр, но добавить подписку на изменения в shared prefs https://developer.android.com/develop/ui/views/components/settings/use-saved-values#changes Т.е. как только произойдут изменения в префах, будет дергаться метод из Арр. Но и так тоже ок)
